src/platform/Zephyr/ConfigurationManagerImpl.cpp

```diff
@@ -181,6 +181,12 @@ void ConfigurationManagerImpl::DoFactoryReset(intptr_t arg)
 {
     ChipLogProgress(DeviceLayer, "Performing factory reset");
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
+    ThreadStackMgr().WaitOnSrpClearAllComplete();
```

Comment:
Why isn't `ThreadStackMgr().ClearAllSrpHostAndServices()` enough? The other functions are called under the hood.

Reply:
I didn't check that, my bad